src/iframe-helper.js

```diff
@@ -416,8 +416,12 @@ export class SubscriptionApi {
    * @param {boolean} is3p set to true if the iframe is 3p.
    * @param {function(!JsonObject, !Window, string)} requestCallback Callback
    *     invoked whenever a new window subscribes.
```

Comment:
>  it would allow the JS within the frame to be treated as friendly with the JS in the parent page such that it could alter/read state. 

i don't think that is true. `allow-same-origin` is just to reserve the iframe origin so that any other frames with same origin can access. cross origin access will never be enabled.